src/slam_karto.cpp

```diff
@@ -280,8 +293,21 @@ SlamKarto::SlamKarto() :
     mapper_->setParamUseResponseExpansion(use_response_expansion);
 
   // Set solver to be used in loop closure
-  solver_ = new SpaSolver();
-  mapper_->SetScanSolver(solver_);
+  try
```

Comment:
To make it more flexible to add new solvers, I have replaced our explicit  if/else if /.. using concrete instantiations ( https://github.com/savioke/slam_karto/blob/savioke-indigo-devel/src/slam_karto.cpp#L384-L431 ) by this. It requires that the 'solver_type' parameter be set to the exact name of the plugin in the `karto_plugins::` namespace but that should not be an issue.